lmcache/storage_backend/serde/cachegen_basics.py

```diff
@@ -39,40 +43,43 @@ def from_model_name(model_name: str) -> "CacheGenConfig":
         family_9b = ["THUDM/glm-4-9b-chat"]
         if model_name in family_7b:
             return CacheGenConfig(
-                key_first_layers=10,
-                key_second_layers=20,
-                key_third_layers=32,  # total layers
-                key_first_bins=32,
-                key_second_bins=16,
-                key_third_bins=16,
-                value_first_layers=2,
-                value_first_bins=32,
-                value_second_bins=16,
+                nlayers=32,
+                kspecs=[
+                    QuantizationSpec(start_layer=0, end_layer=10, bins=32),
+                    QuantizationSpec(start_layer=10, end_layer=20, bins=16),
+                    QuantizationSpec(start_layer=20, end_layer=32, bins=16),
+                ],
+                vspecs=[
+                    QuantizationSpec(start_layer=0, end_layer=2, bins=32),
+                    QuantizationSpec(start_layer=2, end_layer=32, bins=16),
+                ],
             )
         elif model_name in family_8b:
             return CacheGenConfig(
-                key_first_layers=10,
-                key_second_layers=20,
-                key_third_layers=32,  # total layers
-                key_first_bins=32,
-                key_second_bins=16,
-                key_third_bins=16,
-                value_first_layers=2,
-                value_first_bins=32,
-                value_second_bins=16,
+                nlayers=32,
+                kspecs=[
+                    QuantizationSpec(start_layer=0, end_layer=10, bins=32),
+                    QuantizationSpec(start_layer=10, end_layer=20, bins=16),
+                    QuantizationSpec(start_layer=20, end_layer=32, bins=16),
+                ],
+                vspecs=[
+                    QuantizationSpec(start_layer=0, end_layer=2, bins=32),
+                    QuantizationSpec(start_layer=2, end_layer=32, bins=16),
+                ],
             )
         # TODO(Jiayi): needs tuning for better quality
         elif model_name in family_9b:
             return CacheGenConfig(
-                key_first_layers=10,
-                key_second_layers=20,
-                key_third_layers=40,
-                key_first_bins=32,
-                key_second_bins=16,
-                key_third_bins=16,
-                value_first_layers=2,
-                value_first_bins=32,
-                value_second_bins=16,
+                nlayers=40,
+                kspecs=[
+                    QuantizationSpec(start_layer=0, end_layer=10, bins=32),
+                    QuantizationSpec(start_layer=10, end_layer=20, bins=16),
+                    QuantizationSpec(start_layer=20, end_layer=40, bins=16),
+                ],
+                vspecs=[
+                    QuantizationSpec(start_layer=0, end_layer=2, bins=32),
+                    QuantizationSpec(start_layer=2, end_layer=40, bins=16),
+                ],
```

Comment:
Is there any support for the default setting (for models other than these five models)? For exmple, we can pass the model_config in vllm (which contains info about the number of layers) or the number of layers directly as the function param, and configure the `CacheGenConfig` dynamically based on this input.